### PR TITLE
docs: Add smbios docs

### PR DESCRIPTION
**Pull request by Itxaka**
Adds smbios docs and examples

Signed-off-by: Itxaka <igarcia@suse.com>

### CODE REVIEW

File: docs/smbios.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	Product Name: Awesome PC
+	Version: Not Specified
+	Serial Number: THX1138
+	Family: Toretto
+```
+
+And setting the `#!yaml machineName` to `serial-${System Information/Serial Number}` would result in the final value of `serial-THX1138`
+
+This is useful to generate automatic names for machines based on their hardware values, for example using the UUID or the Product name.
+Our default `#!yaml machineName` when the registration values are empty is `"m-${System Information/Manufacturer}-${System Information/Product Name}-${System Information/UUID}"`
```
Comment (fgiudici) [inline docs/smbios.md:35]: ok, I shoud remember to change here when I will be ready for https://github.com/rancher/elemental-operator/pull/145 (this is a note to self)

**Review by fgiudici — APPROVED** [review]
Great, quite clear!

**Comment by Itxaka** [comment]
waiting for @nunix feedback as Im not very happy with the docs, they are a bit...amateur IMO and he usually proposes much nicer changes :+1:

**Comment by nunix** [comment]
> waiting for @nunix feedback as Im not very happy with the docs, they are a bit...amateur IMO and he usually proposes much nicer changes 👍

@Itxaka please go ahead, I'll make a PR for updating it upon my return (don't want to block)

**Pull request merged**